File: Daily_Challenge/20230314_LC_958.py
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        q = collections.deque()
        q.append(root)

        while q:
            cur = q.popleft()
            if cur:
                q.append(cur.left)
                q.append(cur.right)
            else:
                while q:
                    if q.popleft():
                        return False
        return True

File: Daily_Challenge/20230314_LC_958.cpp
class Solution {
public:
    bool isCompleteTree(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()){
            TreeNode* cur = q.front();
            q.pop();
            if (cur != nullptr){
                q.push(cur->left);
                q.push(cur->right);
            }
            else{
                while (!q.empty()){
                    TreeNode* p = q.front();
                    q.pop();
                    if (p != nullptr){
                        return false;
                    }
                }
            }
        }
        return true;
    }
};